subsys/pmci/mctp/mctp_i2c_gpio_controller.c

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2025 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include "zephyr/drivers/gpio.h"
+#include "zephyr/pmci/mctp/mctp_i2c_gpio_common.h"
+#include "zephyr/rtio/rtio.h"
+#include <zephyr/sys/__assert.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/uart.h>
+#include <zephyr/pmci/mctp/mctp_i2c_gpio_controller.h>
+#include <crc-16-ccitt.h>
```

Comment:
Nit: I notice MCTP UART uses CRCs. I wonder if we should use it here as well.

I get the value in UART (async) vs I2C (sync), but maybe this could also help the controller detect SDA corruption due to I2C targets with the same address?